cern_access/indico_cern_access/plugin.py

```diff
@@ -0,0 +1,237 @@
+
+    def _get_access_status(self, regform, registration=None, header=False):
+        if regform.cern_access_request and regform.cern_access_request.is_active:
+            return render_plugin_template('cern_access_status.html',
+                                          registration=registration,
+                                          header=header)
+
+    def _registration_deleted(self, registration, **kwargs):
+        """Withdraws CERN access request for deleted registrations"""
+        if registration.cern_access_request:
+            send_adams_delete_request([registration])
+            registration.cern_access_request.request_state = CERNAccessRequestState.withdrawn
+
+    def _event_time_changed(self, sender, obj, **kwargs):
+        """Updates event time in CERN access requests in ADaMS"""
+        event = obj
+        registrations = get_requested_registrations(event=event)
+        if registrations:
+            state = send_adams_post_request(event, registrations, update=True)[0]
+            if state == CERNAccessRequestState.accepted:
+                update_access_requests(registrations, state)
+
+    def _registration_form_deleted(self, registration_form):
+        """Withdraws CERN access request for deleted registration form and corresponding registrations"""
+        if registration_form.cern_access_request and registration_form.cern_access_request.is_active:
+            registrations = get_requested_registrations(registration_form.event, regform=registration_form)
+            if registrations:
+                deleted = send_adams_delete_request(registrations)
+                if deleted:
+                    withdraw_access_requests(registrations)
+                    notify_access_withdrawn(registrations)
+            registration_form.cern_access_request.request_state = CERNAccessRequestState.withdrawn
+
+    def _event_deleted(self, event, user):
+        """Withdraws CERN access request for registration forms and corresponding registrations of deleted event"""
+        access_requests_forms = get_requested_forms(event)
+        if access_requests_forms:
+            requested_registrations = get_requested_registrations(event=event)
+            if requested_registrations:
+                deleted = send_adams_delete_request(requested_registrations)
+                if deleted:
+                    withdraw_access_requests(requested_registrations)
+                    notify_access_withdrawn(requested_registrations)
+            for form in access_requests_forms:
+                form.cern_access_request.request_state = CERNAccessRequestState.withdrawn
+
+    def _registration_modified(self, registration, change):
+        """If name of registration changed, updates the ADaMS CERN access request"""
+        if registration.cern_access_request and ('first_name' in change or 'last_name' in change):
+            state, _ = send_adams_post_request(registration.event, [registration], update=True)
```

Comment:
`state = ...()[0]` (like above)

If you prefer to use tuple unpacking, use `__` for the unused arg to avoid shadowing the i18n `_`, and change it above too to be consistent